src/streaming/controllers/BlacklistController.js

```diff
@@ -37,10 +37,35 @@ function BlackListController(config) {
     config = config || {};
     let instance;
     let blacklist = [];
+    let blacklistExpiryTimers = [];
 
     const eventBus = EventBus(this.context).getInstance();
+    const settings = config.settings;
     const updateEventName = config.updateEventName;
     const addBlacklistEventName = config.addBlacklistEventName;
+    const unBlacklistEventName = config.unBlacklistEventName;
+
+    function remove(entry) {
+        const index = blacklist.indexOf(entry);
+        if (index >= 0) {
+            blacklist.splice(index, 1);
+
+            eventBus.trigger(
+                unBlacklistEventName, {
+                    entry: entry
+                }
+            );
+        }
+    }
+
+    function setupBlacklistExpiry(entry) {
+        let blacklistExpiryTime = settings && settings.get().streaming.blacklistExpiryTime || 0;
```

Comment:
We made the decision to use -1 as other setting values seem to use that pattern as well.

Reply:
That's fair! Where I've seen `-1` in use it is to allow `0` as a default value and avoid the `|| 0` fallback. On L62 the default `|| 0` will never be used, because `-1` isn't falsey. In other places I've seen `-1` used they use a ternary like `blacklistExpiryTime === -1 ? SOME_DEFAULT : blacklistExpiryTime`

Reply:
Changed the default to 0, I am happy with that 👍